File: Hilbert - Unity/Assets/Scripts/Stone.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Stone : MonoBehaviour
{
    public int xRow;
    public int xCol;
    public bool xIsA;
    public bool xIsAlive = true;
    public bool xIsPlaced = false;
    public bool xIsActive = true;
    public Sprite xStoneImage;
    public bool xBoosted = false;
    public bool xBlocked = false;

    [Space(20)]
    public string xName;
    public string xTypeCell = "Standard";
    public int xHealth = 1;
    public int xAttackPower = 0;
    public int xScorePerTurn = 0;
    public int xMovementHorizontal = 0;
    public int xMovementVertical = 0;
    public int xMovementDiagonal = 0;
    
    [Space(20)]
    public bool xCanMoveOnlyForward = false;
    public bool xCanMoveFromThrone = true;
    public bool xScoreExponentialGrowth = false;
    public bool xSimpleMergePossible = false;
    public int xMergeMultiplier = 0;
    public bool xThroneExplosion = false;
    public int xThroneExplosionSteal = 0;

    [Space(20)]
    public int xHowLongInThrone = 0;
    public int xNewScorePerTurn;

    void Awake()
    {
        if (xName == "Iron")
        {
            xTypeCell = "Guardian";
            xHealth = 50;
            xMovementHorizontal = 2;
            xMovementVertical = 3;
        }

        else if (xName == "Aquamarine")
        {
            xTypeCell = "Luxury";
            xHealth = 5;
            xMovementHorizontal = 2;
            xMovementVertical = 2;
            xScorePerTurn = 30;
        }

        else if (xName == "Jadeite")
        {
            xTypeCell = "Luxury";
            xMovementHorizontal = 2;
            xMovementVertical = 1;
            xHealth = 15;
            xScorePerTurn = 20;
            xMergeMultiplier = 3;
            xSimpleMergePossible = true;

        }

        else if (xName == "Opal")
        {
            xTypeCell = "Luxury";
            xMovementHorizontal = 1;
            xCanMoveFromThrone = false;
            xCanMoveOnlyForward = true;
            xScorePerTurn = 10;
            xScoreExponentialGrowth = true;

        }

        else if (xName == "Adrovay")
        {
            xTypeCell = "Punisher";
            xHealth = 25;
            xMovementHorizontal = 2;
            xMovementVertical = 2;
            xMovementDiagonal = 2;
            xAttackPower = 5;
            xThroneExplosion = true;
            xThroneExplosionSteal = 50;
        }

        else if(xName == "Amethyst")
        {
            xTypeCell = "Punisher";
            xMovementDiagonal = 1;
            xAttackPower = 100;
            xHealth = 5;
        }

    }



}

File: Hilbert - Unity/Main/Scripts/internal/stone.py
import UnityEngine as ue

StoneData = {
    "xRow" : 0, "xCol" : 0, "xIsA" : False, "xIsAlive" : True, "xIsPlaced" : False, "xIsActive" : True, "xStoneImage" : ue.Image(),
    "xBoosted" : False, "xBlocked" : False, "xName" : "defaultName",
    "xTypeCell" : "Standard", "xHealth" : 1, "xAttackPower" : 0, "xScorePerTurn" : 0, "xMovementHorizontal" : 0,
    "xMovementVertical" : 0, "xMovementDiagonal" : 0, "xCanMoveOnlyForward" : False, "xCanMoveFromThrone" : True, "xScoreExponentialGrowth" : False,
    "xSimpleMergePossible" : False, "xMergeMultiplier" : 0, "xThroneExplosion" : False, "xThroneExplosionSteal" : 0
    }
'''
1. Class used for defining every stone by certain properties.
2. This will also be used to manipulate stone's properties
'''
class stone():
    def __init__(self, StoneData):
        self.StoneDat["xRow"] = StoneData["xRow"]
        self.StoneData["xCol"] = StoneData["xCol"]
        self.StoneData["xIsA"] = StoneData["xIsA"]
        self.StoneData["xIsAlive"] = StoneData["xIsAlive"]
        self.StoneData["xIsPlaced"] = StoneData["xIsPlaced"
        self.StoneData["xIsPlaced"] = StoneData["xIsPlaced"]
        self.StoneData["xStoneImage"] = StoneData["xStoneImage"]
        self.StoneData["xBoosted"] = StoneData["xBoosted"]
        self.StoneData["xBlocked"] = StoneData["xBlocked"]
        self.StoneData["xName"] = StoneData["xName"]
        self.StoneData["xTypeCell"] = StoneData["xTypeCell"]
        self.StoneData["xHealth"] = StoneData["xHealth"]
        self.StoneData["xAttackPower"] = StoneData["xAttackPower"]
        self.StoneData["xScorePerTurn"] = StoneData["xScorePerTurn"]
        self.StoneData["xMovementHorizontal"] = StoneData["xMovementHorizontal"]
        self.StoneData["xMovementVertical"] = StoneData["xMovementVertical"]
        self.StoneData["xMovementDiagonal"] = StoneData["xMovementDiagonal"]
        self.StoneData["xCanMoveOnlyForward"] = StoneData["xCanMoveOnlyForward"]
        self.StoneData["xCanMoveFromThrone"] = StoneData["xCanMoveFromThrone"]
        self.StoneData["xScoreExponentialGrowth"] = StoneData["xScoreExponentialGrowth"]
        self.StoneData["xSimpleMergePossible"] = StoneData["xSimpleMergePossible"]
        self.StoneData["xMergeMultiplier"] = StoneData["xMergeMultiplier"]
        self.StoneData["xThroneExplosion"] = StoneData["xThroneExplosion"]
        self.StoneData["xThroneExplosionSteal"] = StoneData["xThroneExplosionSteal"]
